source/common/router/config_impl.cc

```diff
@@ -654,6 +655,21 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
               "not be stripped: {}",
               path_redirect_);
   }
+  if (!route.stat_prefix().empty()) {
+    Stats::StatNameManagedStorage route_stat_name_storage =
+        Stats::StatNameManagedStorage(route.stat_prefix(), factory_context.scope().symbolTable());
+    Stats::ScopeSharedPtr route_stats_scope = Stats::Utility::scopeFromStatNames(
+        *factory_context.scope().scopeFromStatName(
```

Comment:
I don't think we need to make a sub-scope here. Just make one flat vector with the names you want for route_stats_scope.
```
Stats::ScopeSharedPtr route_stats_scope = Stats::Utility::scopeFromStatNames(
    factory_context.scope(), {
        stat_names.vhost_, vhost.statName(), stat_names.route_, route_stat_name_storage_.statName()
    });
```
also per that example make a temp for `factory_context.routerContext().routeStatNames()` for readability.